### PR TITLE
WIP: Extensions

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

**Comment by josevalim** [comment]
Nice work so far!

File: lib/postgrex/extensions/text.ex [hunk 1]
```diff
@@ -0,0 +1,16 @@
+defmodule Postgrex.Extensions.Text do
+  use Behaviour
```
Comment (josevalim) [inline lib/postgrex/extensions/text.ex]: `@behaviour Postgrex.Extensions` you mean?

File: mix.exs [hunk 2]
```diff
@@ -19,7 +19,7 @@ defmodule Postgrex.Mixfile do
 
   # Configuration for the OTP application
   def application do
-    []
+    [applications: [:logger]]
```
Comment (josevalim) [inline mix.exs]: Do we even use Logger?

File: lib/postgrex/extensions/text.ex (hunk 1 above)
Comment (ericmj) [inline lib/postgrex/extensions/text.ex]: Yes! Thank you.

File: mix.exs (hunk 2 above)
Comment (ericmj) [inline mix.exs]: No, but formatting error reports is useful in development and testing. Should we set it based on `Mix.env`?

Comment (josevalim) [inline mix.exs]: Nah, it is fine. If you want you can also start it in your test_helper but it is fine.

**Pull request merged**